src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -117,6 +120,12 @@ object Producer {
         classVersion = data.hashCode.toHexString
       )
   }
+
+  sealed class Source
```

Comment:
* Let's combine the timing and the source into one object.  RuntimeStatistics?  ExecutionInfo?
* The `Producer.copy` method should copy the timing/source data object as well
* The timing/source object should only be overwritten if its value is `Unknown`  
* I don't think the `Cached` source is useful.  You'd rather know what happened when `create` was first called, not the most recent call to `get`